PayloadPkg/OsLoader/BootParameters.c

```diff
@@ -261,6 +262,17 @@ UpdateOsParameters (
   //
   CmdFile = &LoadedImage->Image.Common.CmdFile;
 
+  if ((CurrentBootOption->BootFlags & BOOT_FLAGS_MENDER) != 0 && (CurrentBootOption->BootFlags & BOOT_FLAGS_MISC) == 0) {
+    AsciiSPrint (MenderParams, sizeof (MenderParams), " root=PARTLABEL=primary ");
+    if ((AsciiStrSize (MenderParams) + CmdFile->Size) < CMDLINE_LENGTH_MAX) {
+      AsciiStrCatS ((CHAR8 *)CmdFile->Addr, CMDLINE_LENGTH_MAX, &MenderParams[0]);
+    } else {
+      DEBUG ((DEBUG_INFO, "Unable to append Mender cmd line parameters!\n"));
```

Comment:
Do you consider this error as fatal ?   If so, should we block the boot ?

Reply:
It is not necessary to check BOOT_FLAGS_MISC.
If any of BOOT_FLAGS_MISC and BOOT_FLAGS_MENDER is set, SBL will read data to decide boot slot. the config data tool could only select one of them.

Reply:
Maybe we could put this code to SblBootParameter.c, and refer below code to make minor change.
  //
  // Add OS A/B boot info parameter
  //
  if ((BootOption->BootFlags & BOOT_FLAGS_MISC) != 0) {
    if ((BootOption->BootFlags & LOAD_IMAGE_FROM_BACKUP) == 0) {
      AsciiStrCatS (CommandLine, MaxCmdSize, " suffix=0");
    } else {
      AsciiStrCatS (CommandLine, MaxCmdSize, " suffix=1");
    }
  }